include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,324 @@
+                {
+                    if (kif[i].fd_fd == KERN_FILE_TEXT && 
+                        (st.st_dev == static_cast<dev_t>(kif[i].va_fsid) || 
+                        st.st_ino == static_cast<ino_t>(kif[i].va_fileid))) 
+                    {
+                        *out = executable;
+                        success = true;
+                        break;
+                    }
+                }
+            }
+            kvm_close(kd);
+        }
+        if (*out.empty())
+            ec = detail::get_last_error();
+        return success;
+    };
+    std::vector<std::string> buffer = cmdline_from_proc_id(pid, ec);
+    if (!buffer.empty()) 
+    {
+        bool is_exe = false;
+        std::string argv0;
+        if (!buffer[0].empty()) 
+        {
+            if (buffer[0][0] == '/') 
+            {
+                argv0 = buffer[0];
+                is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+            } 
+            else if (buffer[0].find('/') == std::string::npos) 
+            {
+                std::string penv = envvar_value_from_proc_id(pid, "PATH");
+                if (!penv.empty()) {
+                    std::vector<std::string> env;
+                    std::string tmp;
+                    std::stringstream sstr(penv); 
+                    while (std::getline(sstr, tmp, ':')) 
+                    {
+                        env.push_back(tmp);
+                    }
+                    for (std::size_t i = 0; i < env.size(); i++) 
+                    {
+                        argv0 = env[i] + "/" + buffer[0];
+                        is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+                        if (is_exe) break;
+                        if (buffer[0][0] == '-') 
+                        {
+                            argv0 = env[i] + "/" + buffer[0].substr(1);
+                            is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+                            if (is_exe) break;
+                        }
+                    }
+                }
+            } else {
+                std::string pwd = envvar_value_from_proc_id(pid, "PWD", ec);
+                if (!pwd.empty()) 
+                {
+                    argv0 = pwd + "/" + buffer[0];
+                    is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+                }
+                if (pwd.empty() || !is_exe) 
+                {
+                    std::string cwd = cwd_from_proc_id(pid, ec);
+                    if (!cwd.empty()) 
+                    {
+                       argv0 = cwd + "/" + buffer[0];
+                       is_exe = is_executable(pid, argv0.c_str(), &path, ec);
+                    }
+                }
+            }
+        }
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif defined(__sun)
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
```

Comment:
why the intermediate `path` string btw.? WHy not just return `exe`?

Reply:
I see now, the problem you had earlier with my realpath error handling was related to the fact i am copying the strings, which is more memory being allocated when it could be avoided. I'll fix that, nice catch!